plugins/out_kinesis_streams/kinesis_api.c

```diff
@@ -97,6 +97,31 @@ static int init_put_payload(struct flb_kinesis *ctx, struct flush *buf,
     return -1;
 }
 
+/*
+ * Adjust the time_key output format according to time_key_format.
+ */
+static void idontknow(char *str, size_t maxsize,
+                      const char *time_key_format, const struct tm *timeptr,
+                      size_t len, struct flb_time *tms){
+    size_t s;
+    const char *format = "%Y-%m-%dT%H:%M:%S";
+    len = strftime(str, maxsize, format, timeptr);
+    format = strstr(time_key_format, ".%F");
```

Comment:
why is the user required to put a dot `.` character in there? Can we support user provided timestamp formats where the milliseconds do not include a dot?

Reply:
In my opinion, customers can add millisecond or nanosecond with time_key_format %Y-%m-%dT%H:%M:%S.%F or %Y-%m-%dT%H:%M:%S.%L.  because default setting is %Y-%m-%dT%H:%M:%S so I put a dot `.` here. Of course we can delete this dot in logic.

Reply:
The goal of time_key_format is to give the user full control over the timestamp format. If we require that milliseconds or nanoseconds are prefaced with a dot then we are restricting the user unnecessarily IMO. Just because that might be the most common or obvious use case does not mean it is the only one.